### ISSUE TITLE
Allow using docker for Darwin targets

### ISSUE THREAD

**Issue by rpignolet** [rust-embedded/cross#436]
It is related to #223 but I'm not asking to support Darwin targets, just allow to use docker for Darwin targets from Linux host.

I build myself a docker container which support cross compile from Linux to Darwin but I'm unable to use it with this Cross.toml :
```
[target.x86_64-apple-darwin]
image = "custom-cross:x86_64-apple-darwin"
```
When I run `cross build --target x86_64-apple-darwin --release -v` I don't see the `docker run` command in the log, `cross` fallback to `cargo` on the host.

**Comment by UebelAndre** [comment]
Seems like this work is done? Also @rpignolet would you be willing to share your dockerfile?

**Comment by rpignolet** [comment]
I build the file `MacOSX10.15.sdk.tar.xz` with `osxcross` and then use this Dockerfile.

``` dockerfile
FROM ubuntu:18.04

COPY common.sh /
RUN /common.sh

COPY cmake.sh /
RUN /cmake.sh

COPY xargo.sh /
RUN /xargo.sh

RUN mkdir /opt/osxcross &&                                      \
    cd /opt &&                                                  \
    git clone https://github.com/tpoechtrager/osxcross.git &&   \
    cd osxcross &&                                              \
    git checkout c2ad5e859d12a295c3f686a15bd7181a165bfa82 &&    \
    ./tools/get_dependencies.sh

ENV OSXCROSS_SDK_VERSION 10.15

COPY MacOSX${OSXCROSS_SDK_VERSION}.sdk.tar.xz /opt/osxcross/tarballs

RUN cd /opt/osxcross && \
    SDK_VERSION=$OSXCROSS_SDK_VERSION UNATTENDED=1 ./build.sh

#RUN cd /opt/osxcross && ./build_compiler_rt.sh

ENV CARGO_TARGET_X86_64_APPLE_DARWIN_LINKER=/opt/osxcross/target/bin/o64-clang  \
    CC_x86_64_apple_darwin=/opt/osxcross/target/bin/o64-clang                   \
    CXX_x86_64_apple_darwin=/opt/osxcross/target/bin/o64-clang++
```

**Comment by rpignolet** [comment]
Issue is resolved with #489.
